pkg/queue/health_server.go

```diff
@@ -0,0 +1,90 @@
+/*
```

Comment:
Could you update the description to include changes proposed in this PR? Because instead of only adding Prometheus metrics, there're also some other refactoring change.

Reply:
Yeah will do.

Reply:
Sorry for the delay in respone, it's been a long day. Let me know if the description is adequate enough now at your earliest convenience.

Reply:
LGTM. Thanks.